### PR TITLE
add task solution

### PR DESCRIPTION
**Pull request by pawelnowicki87**
- [DEMO LINK](https://pawel.nowicki87.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://pawel.nowicki87.github.io/layout_product-cards/report/html_report/)

- [x] there 2 BEM blocks `card` and `stars` each in it own file
- [x] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [x] SCSS Variables are used for main values and placed in a **separate file**
- [x] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

### CODE REVIEW

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code.
Also, you need to add readme.md file with links to the next commit

**Review by nazarmatsevych — CHANGES_REQUESTED** [review]
The Code looks good but please push readme file or your links in the pull request description.
Make sure to run "npm run deploy" command to deploy your changes

**Comment by pawelnowicki87** [comment]
- [DEMO LINK](https://pawel.nowicki87.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://pawel.nowicki87.github.io/layout_product-cards/report/html_report/)

- [x] there 2 BEM blocks `card` and `stars` each in it own file
- [x] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [x] SCSS Variables are used for main values and placed in a **separate file**
- [x] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

File: src/index.html [hunk 1]
```diff
@@ -1,13 +1,50 @@
...
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;1,700&display=swap" rel="stylesheet">
   </head>
+
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img class="card__product-img" src="./images/imac.jpeg"
+      alt="monitor">
```
Comment (volodymyr-soltys97) [inline src/index.html:19]:
```suggestion
      <img
        class="card__product-img"
        src="./images/imac.jpeg"
        alt="iMac"
      >
```

File: src/index.html [hunk 2]
```diff
@@ -1,13 +1,50 @@
...
+      <div class="card__product-name">
+        <p>APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)</p>
+      </div>
```
Comment (volodymyr-soltys97) [inline src/index.html:22]:
```suggestion
      <h2 class="card__product-name">
        APPLE A1419 iMac 27" Retina 5K Monoblock (MNED2UA/A)
      </h2>
```

File: src/index.html [hunk 3]
```diff
@@ -1,13 +1,50 @@
...
+      <div class="card__product-code">
+        <p>Product code: 195434</p>
+      </div>
```
Comment (volodymyr-soltys97) [inline src/index.html:25]: The same as above, need to fix it everywhere

File: src/index.html [hunk 4]
```diff
@@ -1,13 +1,50 @@
...
+        <div class="card__product-priceText">
+          <p>Price:</p>
+        </div>
+        <div class="card__product-priceValue">
+          <p>&dollar;2,199</p>
+        </div>
+      </div>
+      <a class="card__purchase-link"
+        data-qa="hover"
+        href="#">BUY</a>
```
Comment (volodymyr-soltys97) [inline src/index.html:46]:
```suggestion
      <a
        class="card__purchase-link"
        data-qa="hover"
        href="#"
      >
        BUY
      </a>
```

File: src/styles/blocks/card.scss [hunk 5]
```diff
@@ -0,0 +1,86 @@
...
+    font-weight: 500;
+    line-height: 18px;
+  }
+
+  &__product-code {
+    margin-bottom: 16px;
+    min-height: 14px;
+    color: $secondary-accent;
+    font-size: 10px;
+    font-weight: 400;
```
Comment (volodymyr-soltys97) [inline src/styles/blocks/card.scss:34]: Default value, so you can just remove it everywhere
```suggestion
```

File: src/styles/blocks/card.scss [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+    line-height: 18px;
+  }
+
+  &__purchase-link {
+    display: flex;
+    justify-content: center;
+    text-align: center;
+    align-items: center;
+    margin-bottom: 16px;
+    width: 166px;
```
Comment (volodymyr-soltys97) [inline src/styles/blocks/card.scss:65]:
```suggestion
    width: 100%;
```

File: src/styles/blocks/stars.scss [hunk 7]
```diff
@@ -0,0 +1,33 @@
+@import "../utils/colors";
+
+.stars {
+  margin-bottom: 16px;
+  width: 166px;
```
Comment (volodymyr-soltys97) [inline src/styles/blocks/stars.scss:5]: The same as above, no need to hardcode width in px

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Great!
Let's improve your code

**Comment by pawelnowicki87** [comment]
When i add quotes arount font family "Roboto" npm in vscode show it is an error, but when i change it on single quote 'Roboto' then it passes test, but on github it wants to change it again to double quotes. So how it should be?

File: readme.md [hunk 8]
```diff
@@ -20,8 +20,8 @@ Create a pages with product card using `flexbox`, `BEM` and `SCSS` based on [thi
 
 ❗️ Replace `<your_account>` with your Github username and copy the links to `Pull Request` description:
 
-- [DEMO LINK](https://<your_account>.github.io/layout_product-cards/)
-- [TEST REPORT LINK](https://<your_account>.github.io/layout_product-cards/report/html_report/)
+- [DEMO LINK](https://pawel.nowicki87.github.io/layout_product-cards/)
```
Comment (anastasiiavorobiova) [inline readme.md:23]:
```suggestion
- [DEMO LINK](https://pawelnowicki87.github.io/layout_product-cards/)
```

File: readme.md [hunk 9]
```diff
@@ -20,8 +20,8 @@ Create a pages with product card using `flexbox`, `BEM` and `SCSS` based on [thi
...
+- [TEST REPORT LINK](https://pawel.nowicki87.github.io/layout_product-cards/report/html_report/)
```
Comment (anastasiiavorobiova) [inline readme.md:24]:
```suggestion
- [TEST REPORT LINK](https://pawelnowicki87.github.io/layout_product-cards/report/html_report/)
```

File: src/index.html [hunk 10]
```diff
@@ -1,13 +1,52 @@
 <!doctype html>
+
 <html lang="en">
+
   <head>
     <meta charset="UTF-8">
     <meta name="viewport"
```
Comment (anastasiiavorobiova) [inline src/index.html:7]:
```suggestion
    <meta 
      name="viewport"
```

File: src/index.html [hunk 11]
```diff
@@ -1,13 +1,52 @@
...
-          content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
+      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0">
```
Comment (anastasiiavorobiova) [inline src/index.html:8]:
```suggestion
      content="width=device-width, user-scalable=no, initial-scale=1.0, maximum-scale=1.0, minimum-scale=1.0"
    >
```

File: src/index.html [hunk 12]
```diff
@@ -1,13 +1,52 @@
...
+    <link rel="preconnect" href="https://fonts.googleapis.com">
+    <link rel="preconnect" href="https://fonts.gstatic.com">
+    <link href="https://fonts.googleapis.com/css2?family=Roboto:ital,wght@0,400;0,500;1,700&display=swap" rel="stylesheet">
   </head>
+
   <body>
-    <h1>Product cards</h1>
+    <div class="card" data-qa="card">
+      <img class="card__product-img" src="./images/imac.jpeg"
+      alt="iMac">
```
Comment (anastasiiavorobiova) [inline src/index.html:19]:
```suggestion
      <img 
        class="card__product-img" 
        src="./images/imac.jpeg"
        alt="iMac"
      >
```
Consider fixing indents according to the code style guide

File: src/index.html [hunk 13]
```diff
@@ -1,13 +1,52 @@
...
+      <h2 class="card__product-code">
+        Product code: 195434
+      </h2>
+      <div class="stars stars--4">
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <div class="stars__star"></div>
+        <h2 class="stars__reviews">
```
Comment (anastasiiavorobiova) [inline src/index.html:32]:
```suggestion
        <p class="stars__reviews">
```

File: src/index.html [hunk 14]
```diff
@@ -1,13 +1,52 @@
...
+          Reviews: 5
+        </h2>
+      </div>
+      <div class="card__product-price">
+        <h2 class="card__product-priceText">
```
Comment (anastasiiavorobiova) [inline src/index.html:37]:
```suggestion
        <p class="card__product-priceText">
```

File: src/index.html [hunk 15]
```diff
@@ -1,13 +1,52 @@
...
+          Price:
+        </h2>
+        <h2 class="card__product-priceValue">
```
Comment (anastasiiavorobiova) [inline src/index.html:40]:
```suggestion
        <p class="card__product-priceValue">
```

File: src/index.html [hunk 16]
```diff
@@ -1,13 +1,52 @@
...
+          &dollar;2,199
+        </h2>
+      </div>
+      <a class="card__purchase-link"
+        data-qa="hover"
+        href="#">
+        BUY
```
Comment (anastasiiavorobiova) [inline src/index.html:47]:
```suggestion
      <a 
        class="card__purchase-link"
        data-qa="hover"
        href="#"
      >
        BUY
```

File: src/styles/main.scss [hunk 17]
```diff
@@ -0,0 +1,12 @@
+body {
+  margin: 0;
+  font-family: "Roboto", sans-serif;
+
+  h2 {
```
Comment (anastasiiavorobiova) [inline src/styles/main.scss:5]:
```suggestion
  h2, p {
```

File: src/styles/main.scss [hunk 18]
```diff
@@ -0,0 +1,12 @@
...
+    margin: 0;
+  }
+}
+
```
Comment (anastasiiavorobiova) [inline src/styles/main.scss:9]:
```suggestion
* {
  box-sizing: border-box;
}
```

File: src/styles/blocks/card.scss [hunk 19]
```diff
@@ -0,0 +1,85 @@
+@import "../utils/colors";
+
+.card {
+  box-sizing: border-box;
```
Comment (anastasiiavorobiova) [inline src/styles/blocks/card.scss:4]:
```suggestion

```

**Review by anastasiiavorobiova — CHANGES_REQUESTED** [review]
Great work! Almost done!

**Comment by pawelnowicki87** [comment]
- [DEMO LINK](https://pawelnowicki87.github.io/layout_product-cards/)
- [TEST REPORT LINK](https://pawelnowicki87.github.io/layout_product-cards/report/html_report/)

❗️ Copy this `Checklist` to the `Pull Request` description after links, and put `- [x]` before each point after you checked it.

- [x] there 2 BEM blocks `card` and `stars` each in it own file
- [x] SCSS Nesting is used for `elements`, `modifiers` and `pseudo-classes`
- [x] SCSS Variables are used for main values and placed in a **separate file**
- [x] all `stars--N` modifiers work as expected (Highlight first `N` stars)
- [x] Code follows all the [Code Style Rules ❗️](https://mate-academy.github.io/layout_task-guideline/html-css-code-style-rules)

**Review by volodymyr-soltys97 — CHANGES_REQUESTED** [review]
Good job 👍 
Let's improve your code style from the previous review

**Pull request closed without merging**